File: labs-training-master/java/solid/solid-java-1/src/main/java/es/emergya/labs/training/solid_java_1/ocp/excercise_00/MagicMachine.java
package es.emergya.labs.training.solid_java_1.ocp.excercise_00;

public class MagicMachine {

	public void produce(IngredientsInterface ingredientsPackage) {
		ingredientsPackage.makeFood();
	}
	
	class NoThatMagicException extends RuntimeException {

		private static final long serialVersionUID = 5614958177835014488L;
	}
	
}

File: labs-training-master/SolidPythonResuelto/main/ocp/MagicMachine.py
from HotDogIngredientsPackage import HotdogIngredientsPackage
from PizzaIngredientsPackage import PizzaIngredientsPackage
from IngredientsPackageInterface import IngredientsPackageInterface

class NoThatMagicException(Exception) :
        serialVersionUID = 5614958177835014488

class MagicMachine :
    def produce(self, ingredientsPackage:IngredientsPackageInterface) -> None:
        if (ingredientsPackage == None) :
            raise NoThatMagicException()
        else :
            ingredientsPackage.makeFood()

